### PR TITLE
[Benchmark] Cholesky decomposition benchmark

### PR DESCRIPTION
**Pull request by zzzDavid**
## Description
This PR implements Cholesky decomposition algorithm in Bril. 

Cholesky decomposition transforms a Hermitian, positive-definite matrixinto the product of a  lower triangular matrix and its conjugate transpose. It is useful for numerical methods such as Monte Carlo Simulation or solving systems of linear equations.

### CODE REVIEW

**Review by sampsyo — CHANGES_REQUESTED** [review]
This is a very ambitious benchmark; nice work!

Can you please add a bullet to the list in the documentation describing the benchmark?

**Comment by zzzDavid** [comment]
> This is a very ambitious benchmark; nice work!
> 
> Can you please add a bullet to the list in the documentation describing the benchmark?

Thanks, I added the documentation, also fixed a bug. Now the Cholesky decomposition benchmark gets the same result as `scipy.linalg.cholesky`

**Review by sampsyo — APPROVED** [review]
Looks great; thanks!!

File: benchmarks/cholesky.bril [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+  rval: float = load rvalloc;
+  prod: float = fmul lval rval;
+  sum: float = fadd sum prod;
+.sum.loop_end:
+  i: int = add i one;
+  jmp .sum.loop;
+.sum.done:
+  idx: int = mul row size;
+  idx: int = add idx col;
+  loc: ptr<int> = ptradd dest idx;
```
Comment (Pat-Lafon) [inline benchmarks/cholesky.bril:142]: I think you meant `ptr<float>`

File: benchmarks/cholesky.bril [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+  ij_index: int = add ij_index j;
+  jj_index: int = mul j size;
+  jj_index: int = add jj_index j;
+  a_ij_ptr: ptr<float> = ptradd arr1 ij_index;
+  b_ij_ptr: ptr<float> = ptradd arr2 ij_index;
+  jj_ptr: ptr<float> = ptradd arr2 jj_index;
+  a_ij = load a_ij_ptr;
+  b_ij = load b_ij_ptr;
+  b_jj = load jj_ptr;
+  value = fdiv a_ij b_jj;
```
Comment (Pat-Lafon) [inline benchmarks/cholesky.bril:297]: `:float` ?

File: benchmarks/cholesky.bril (hunk 1 above)
Comment (zzzDavid) [reply benchmarks/cholesky.bril:142]: Thanks, this issue is resolved by this PR: https://github.com/sampsyo/bril/pull/158, please review.

File: benchmarks/cholesky.bril (hunk 2 above)
Comment (zzzDavid) [reply benchmarks/cholesky.bril:297]: This issue is also fixed in PR https://github.com/sampsyo/bril/pull/158

**Pull request merged**